Incompatibility with TypeScript 3.7.2

The newly released TypeScript has added `supportedEntryTypes ` ([source](https://github.com/microsoft/TypeScript/blob/master/lib/lib.webworker.d.ts#L2569)) property to the `PerformanceObserver` instance. But it does not exist in @fastly/performance-observer-polyfill. This will lead to a compilation error.

Currently I have to use `skipLibCheck` to ignore this error. Please support the new version. Thanks very much!

@phamann Sorry that I didn't make myself clear. I was too busy and the issue was opened in hurry.

The problem appears when @fastly/performance-observer-polyfill is used as a dependency, not compiling the lib itself. Here is the minimum reproduction using TypeScript 3.7.2:

index.ts:

```typescript
import "@fastly/performance-observer-polyfill/polyfill";
```

tsconfig.json:

```json
{
  "compilerOptions": {
    "target": "es2015",
    "moduleResolution": "node",
    "strict": true
  },
  "include": [
    "index.ts"
  ]
}
```

The error message:

```
$ npx tsc -p .
node_modules/@fastly/performance-observer-polyfill/polyfill/index.ts:3:31 - error TS2322: Type '{ new (callback: PerformanceObserverCallback): PerformanceObserver; prototype: PerformanceObserver; readonly supportedEntryTypes: readonly string[]; } | typeof PerformanceObserver' is not assignable to type '{ new (callback: PerformanceObserverCallback): PerformanceObserver; prototype: PerformanceObserver; readonly supportedEntryTypes: readonly string[]; }'.
  Property 'supportedEntryTypes' is missing in type 'typeof PerformanceObserver' but required in type '{ new (callback: PerformanceObserverCallback): PerformanceObserver; prototype: PerformanceObserver; readonly supportedEntryTypes: readonly string[]; }'.

3 if (!ctx.PerformanceObserver) ctx.PerformanceObserver = PerformanceObserver;
                                ~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:11778:14
    11778     readonly supportedEntryTypes: ReadonlyArray<string>;
                       ~~~~~~~~~~~~~~~~~~~
    'supportedEntryTypes' is declared here.


Found 1 error.
```

Thanks!

--- a/src/observer.ts
+++ b/src/observer.ts
@@ -1,7 +1,12 @@
 import PerformanceObserverTaskQueue from "./task-queue";
 import EntryList from "./entry-list";
 
-const VALID_TYPES = ["mark", "measure", "navigation", "resource"];
+const VALID_TYPES: readonly string[] = [
+  "mark",
+  "measure",
+  "navigation",
+  "resource"
+];
 const ERRORS = {
   "no-entry-types": `Failed to execute 'observe' on 'PerformanceObserver': required member entryTypes is undefined.`,
   "invalid-entry-types": `Failed to execute 'observe' on 'PerformanceObserver': A Performance Observer MUST have at least one valid entryType in its entryTypes attribute.`
@@ -22,6 +27,7 @@ class PerformanceObserver implements PollingPerformanceObserver {
   public callback: PerformanceObserverCallback;
   public buffer: Set<PerformanceEntry>;
   public entryTypes: string[] = [];
+  public static readonly supportedEntryTypes = VALID_TYPES;
   private taskQueue: PerformanceObserverTaskQueue;
 
   public constructor(

